### PR TITLE
added API reference

### PR DESCRIPTION
**Pull request by ameshkov**
@el-termikael please do the following: review the document and explain how the documentation needs to be changed. Then @atropnikov will update the API spec and I'll re-generate the document.

### CODE REVIEW

**Review by ameshkov — COMMENTED** [review]
Feedback example

File: docs/private-dns/api-reference.md [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+    https://api.adguard-dns.io/static/swagger/openapi.json to markdown using
+    https://swagger-markdown-ui.netlify.app/.
+
+    If you want to change it, ask the developers to change the OpenAPI spec.
+-->
+
+# AdGuard DNS API
+DNS API documentation
+
+## Version: 1.3
```
Comment (atropnikov) [inline docs/private-dns/api-reference.md:19]: Update version to 1.4. API was updated recently (added custom blocking mode option) and I forgot to bump version in repo. Also maybe we need to add CHANGELOG here?

Comment (ameshkov) [reply docs/private-dns/api-reference.md:19]: It's autogenerated from https://api.adguard-dns.io/static/swagger/openapi.json

Comment (ameshkov) [reply docs/private-dns/api-reference.md:19]: Changelog is a good idea. Can you make it available on `api.adguard-dns.io`?

Comment (atropnikov) [reply docs/private-dns/api-reference.md:19]: Done. New version at https://api.adguard-dns.io/static/swagger/openapi.json and [changelog](https://api.adguard-dns.io/static/api/CHANGELOG.md)

File: docs/private-dns/api-reference.md [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+
+### /oapi/v1/account/limits
+
+#### GET
+##### Summary:
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:24]: this subheading seems superfluous
probably, it's better to format the text in this way:

```
#### GET

 Get account limits

 ##### Responses

| Code | Description |
 | ---- | ----------- |
 | 200 | Account limits info |

#### POST

 Creates a new device

##### Responses
<...>
```

File: docs/private-dns/api-reference.md [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+
+Get account limits
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:26]:
```suggestion
Gets account limits
```

File: docs/private-dns/api-reference.md [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+##### Summary:
+
+Get DNS over TLS .mobileconfig.
+
+##### Parameters
+
+| Name | Located in | Description | Required | Schema |
+| ---- | ---------- | ----------- | -------- | ---- |
+| device_id | path |  | Yes | string |
+| exclude_wifi_networks | query | List Wi-Fi networks by their SSID, in which you want AdGuard DNS to be disabled | No | [ string ] |
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:151]: extra comma before 'in which'

File: docs/private-dns/api-reference.md [hunk 5]
```diff
@@ -0,0 +1,561 @@
...
+
+Get DNS over HTTPS .mobileconfig.
+
+##### Parameters
+
+| Name | Located in | Description | Required | Schema |
+| ---- | ---------- | ----------- | -------- | ---- |
+| device_id | path |  | Yes | string |
+| exclude_wifi_networks | query | List Wi-Fi networks by their SSID, in which you want AdGuard DNS to be disabled | No | [ string ] |
+| exclude_domain | query | List domains that will use the default DNS servers instead of AdGuard DNS | No | [ string ] |
```
[the remark this answers is not in the document]
Comment (el-termikael) [reply docs/private-dns/api-reference.md:130]: 'the' is not needed

File: docs/private-dns/api-reference.md [hunk 6]
```diff
@@ -0,0 +1,561 @@
...
+
+##### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 200 | DNS over HTTPS plist file |
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:158]: DNS-over-HTTPS

File: docs/private-dns/api-reference.md [hunk 7]
```diff
@@ -0,0 +1,561 @@
...
+### /oapi/v1/dns_servers
+
+#### GET
+##### Summary:
+
+Lists DNS servers
+
+##### Description:
+
+Lists DNS servers for user. By default exists default DNS server
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:191]: Lists DNS servers for a user

«по умолчанию существует дефолтный днс-сервер» — к чему это?

File: docs/private-dns/api-reference.md [hunk 8]
```diff
@@ -0,0 +1,561 @@
...
+| 200 | List of DNS servers |
+
+#### POST
+##### Summary:
+
+Creates a new DNS server
+
+##### Description:
+
+Creates a new DNS server. You can attach custom settings, otherwise DNS server will be created with default settings.
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:206]:
```suggestion
Creates a new DNS serve that you can customize to your liking.
```
or, more formally:
```suggestion
Creates a new DNS serve that you can customize according to your preferences.
```

File: docs/private-dns/api-reference.md [hunk 9]
```diff
@@ -0,0 +1,561 @@
...
+### /oapi/v1/dns_servers/{dns_server_id}
+
+#### DELETE
+##### Summary:
+
+Removes a DNS server
+
+##### Description:
+
+Removes a DNS server. All devices attached to this DNS server will be moved to the default DNS server. Deletion of default DNS server is forbidden.
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:225]: Deletion of —> Deleting

File: docs/private-dns/api-reference.md [hunk 10]
```diff
@@ -0,0 +1,561 @@
...
+| 200 | DNS server settings updated |
+| 400 | Validation failed |
+| 404 | DNS server not found |
+
+### /oapi/v1/filter_lists
+
+#### GET
+##### Summary:
+
+Get filter lists
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:303]: Gets

File: docs/private-dns/api-reference.md [hunk 11]
```diff
@@ -0,0 +1,561 @@
...
+| Code | Description |
+| ---- | ----------- |
+| 200 | List of filters |
+
+### /oapi/v1/oauth_token
+
+#### POST
+##### Summary:
+
+Generate Access and Refresh token
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:316]: Generates

File: docs/private-dns/api-reference.md [hunk 12]
```diff
@@ -0,0 +1,561 @@
...
+| 401 | Invalid credentials, MFA token or refresh token provided |
+
+null
+
+### /oapi/v1/query_log
+
+#### DELETE
+##### Summary:
+
+Clear query log
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:333]: Clears

File: docs/private-dns/api-reference.md [hunk 13]
```diff
@@ -0,0 +1,561 @@
...
+##### Responses
+
+| Code | Description |
+| ---- | ----------- |
+| 202 | Query log was cleared |
+
+#### GET
+##### Summary:
+
+Get query log
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:344]: Gets

File: docs/private-dns/api-reference.md [hunk 14]
```diff
@@ -0,0 +1,561 @@
...
+| 200 | Refresh token revoked |
+
+null
+
+### /oapi/v1/stats/categories
+
+#### GET
+##### Summary:
+
+Get categories statistics
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:393]: Gets

File: docs/private-dns/api-reference.md [hunk 15]
```diff
@@ -0,0 +1,561 @@
...
+| ---- | ----------- |
+| 200 | Categories statistics received |
+| 400 | Validation failed |
+
+### /oapi/v1/stats/companies
+
+#### GET
+##### Summary:
+
+Get companies statistics
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:416]: Gets

File: docs/private-dns/api-reference.md [hunk 16]
```diff
@@ -0,0 +1,561 @@
...
+| ---- | ----------- |
+| 200 | Companies statistics received |
+| 400 | Validation failed |
+
+### /oapi/v1/stats/companies/detailed
+
+#### GET
+##### Summary:
+
+Get detailed companies statistics
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:439]: Gets

File: docs/private-dns/api-reference.md [hunk 17]
```diff
@@ -0,0 +1,561 @@
...
+| ---- | ----------- |
+| 200 | Detailed companies statistics received |
+| 400 | Validation failed |
+
+### /oapi/v1/stats/countries
+
+#### GET
+##### Summary:
+
+Get countries statistics
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:463]: Gets

File: docs/private-dns/api-reference.md [hunk 18]
```diff
@@ -0,0 +1,561 @@
...
+| ---- | ----------- |
+| 200 | Countries statistics received |
+| 400 | Validation failed |
+
+### /oapi/v1/stats/devices
+
+#### GET
+##### Summary:
+
+Get devices statistics
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:486]: Gets

File: docs/private-dns/api-reference.md [hunk 19]
```diff
@@ -0,0 +1,561 @@
...
+| ---- | ----------- |
+| 200 | Devices statistics received |
+| 400 | Validation failed |
+
+### /oapi/v1/stats/domains
+
+#### GET
+##### Summary:
+
+Get domains statistics
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:509]: Gets

File: docs/private-dns/api-reference.md [hunk 20]
```diff
@@ -0,0 +1,561 @@
...
+| ---- | ----------- |
+| 200 | Domains statistics received |
+| 400 | Validation failed |
+
+### /oapi/v1/stats/time
+
+#### GET
+##### Summary:
+
+Get time statistics
```
Comment (el-termikael) [inline docs/private-dns/api-reference.md:532]: Gets

**Review by el-termikael — CHANGES_REQUESTED** [review]

**Comment by ameshkov** [comment]
@atropnikov can you please let me know when the changes from @el-termikael are available on `api.adguard-dns.io` so that I could re-generate the markdown?

File: docs/private-dns/api-reference.md (hunk 2 above)
Comment (ameshkov) [reply docs/private-dns/api-reference.md]: This is auto-generated, let's keep it as is for the sake of simplicity.

File: docs/private-dns/api-reference.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ameshkov) [reply docs/private-dns/api-reference.md]: Not necessarily, the default is the proper word here to be honest

File: docs/private-dns/api-reference.md (hunk 8 above)
Comment (ameshkov) [reply docs/private-dns/api-reference.md]: Not really, the current description is more precise which is important for an API document.

**Comment by atropnikov** [comment]
> @atropnikov can you please let me know when the changes from @el-termikael are available on `api.adguard-dns.io` so that I could re-generate the markdown?

It's ready.

**Pull request merged**